libraries/botbuilder-dialogs-adaptive/tests/adaptiveDialogBot.test.ts

```diff
@@ -0,0 +1,236 @@
+    let resourceExplorer: ResourceExplorer;
+    let telemetryClient: NullTelemetryClient;
+    let resourceProvider: MockResourceProvider;
+    this.beforeEach(function () {
+        storage = new MemoryStorage();
+        conversationState = new ConversationState(storage);
+        userState = new UserState(storage);
+        skillConversationIdFactory = new SkillConversationIdFactory(storage);
+        languagePolicy = new LanguagePolicy('en-US');
+        resourceExplorer = new ResourceExplorer();
+        telemetryClient = new NullTelemetryClient();
+        resourceExplorer.registerType('Microsoft.AdaptiveDialog', AdaptiveDialog);
+        resourceProvider = new MockResourceProvider(resourceExplorer);
+    });
+
+    it('adds the correct parameters to TurnState', async function () {
+        resourceProvider.add(
+            'main.dialog',
+            new MockResource(
+                JSON.stringify({
+                    $kind: 'Microsoft.AdaptiveDialog',
+                })
+            )
+        );
+        resourceExplorer.addResourceProvider(resourceProvider);
+
+        const activity = {
+            type: ActivityTypes.Message,
+            channelId: 'test-channel',
+            conversation: {
+                id: 'test-conversation-id',
+            },
+            from: {
+                id: 'test-id',
+            },
+        };
+
+        const turnContext = new TurnContext(new TestAdapter(), activity as Activity);
+
+        const bot = new AdaptiveDialogBot(
+            'main.dialog',
+            'main.lg',
+            resourceExplorer,
+            conversationState,
+            userState,
+            skillConversationIdFactory,
+            languagePolicy,
+            new MockBotFrameworkAuthentication() as BotFrameworkAuthentication,
+            telemetryClient
+        );
+
+        await bot.run(turnContext);
+
+        assert.ok(turnContext.turnState.get(BotFrameworkClientKey));
+        assert.ok(turnContext.turnState.get(skillConversationIdFactoryKey));
+        assert.ok(turnContext.turnState.get('ConversationState'));
+        assert.ok(turnContext.turnState.get('UserState'));
+        assert.ok(turnContext.turnState.get(resourceExplorerKey));
+        assert.ok(turnContext.turnState.get(languageGeneratorKey));
+        assert.ok(turnContext.turnState.get(languageGeneratorManagerKey));
+        assert.ok(turnContext.turnState.get(languagePolicyKey));
+        assert.ok(turnContext.turnState.get('memoryScopes'));
+        assert.ok(turnContext.turnState.get('pathResolvers'));
+        assert.ok(turnContext.turnState.get(BotTelemetryClientKey));
+        assert.ok(turnContext.turnState.get(BotCallbackHandlerKey));
+    });
+
+    it('should throw an error when no resource', async function () {
+        resourceExplorer.addResourceProvider(resourceProvider);
+
+        const activity = {
+            type: ActivityTypes.Message,
+            channelId: 'test-channel',
```

Comment:
I'm not saying you _should_, but there is `Channels.Test` that's used in the `TestAdapter`.

https://github.com/microsoft/botbuilder-js/blob/9719123981070d0c63a0090d9f1b8c32de3a41a5/libraries/botframework-schema/src/index.ts#L2155

Otherwise I'm fine with the existing tests as they are 😄